velox/functions/sparksql/tests/ArrayShuffleTest.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/VectorReaders.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+using namespace facebook::velox::test;
+
+class ArrayShuffleTest : public SparkFunctionBaseTest {
+ protected:
+  void testShuffle(
+      const VectorPtr& input,
+      const VectorPtr& expected,
+      int64_t seed,
+      int32_t partitionId = 0) {
+    setSparkPartitionId(partitionId);
+    assertEqualVectors(
+        evaluate(fmt::format("shuffle(c0, {})", seed), makeRowVector({input})),
+        expected);
+  }
+
+  template <typename T>
+  void testShuffle(
+      const VectorPtr& input,
+      const std::vector<std::string>& expected,
+      int64_t seed,
+      int32_t partitionId = 0) {
+    testShuffle(input, makeArrayVectorFromJson<T>(expected), seed, partitionId);
+  }
+
+  template <typename T>
+  void testShuffle(
+      const std::string& input,
+      const std::string& expected,
+      int64_t seed,
+      int32_t partitionId = 0) {
+    testShuffle<T>(
+        makeArrayVectorFromJson<T>({input}), {expected}, seed, partitionId);
+  }
+};
+
+TEST_F(ArrayShuffleTest, basic) {
+  testShuffle<int64_t>("[1, 2, 3, 4, 5]", "[3, 5, 4, 1, 2]", 0);
+  testShuffle<std::string>(
+      "[\"a\", \"b\", \"c\", \"d\"]", "[\"a\", \"c\", \"b\", \"d\"]", 0);
```

Comment:
use R"(["a", "b", "c", "d"])" for readability

Reply:
Thanks.